src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,727 @@
+        /// <param name="parameterValue">paramter value string</param>
+        /// <returns>validate parameter value string</returns>
+        /// <exception cref="ArgumentException">If the paramter value is empty</exception>
+        private static string ValidateQuotedStringOrToken(string parameterValue)
+        {
+            if (String.CompareOrdinal(parameterValue, String.Empty) == 0)
+                throw new ArgumentException(SR.InvalidParameterValue);
+
+            if (parameterValue.Length >= 2 &&
+                parameterValue.StartsWith(Quote, StringComparison.Ordinal) &&
+                parameterValue.EndsWith(Quote, StringComparison.Ordinal))
+                ValidateQuotedText(parameterValue.Substring(1, parameterValue.Length - 2));
+            else
+                ValidateToken(parameterValue);
+
+            return parameterValue;
+        }
+
+        /// <summary>
+        /// This method validates if the text in the quoted string
+        /// </summary>
+        /// <param name="quotedText"></param>
+        private static void ValidateQuotedText(string quotedText)
+        {
+            //empty is okay
+
+            for (int i = 0; i < quotedText.Length; i++)
+            {
+                if (IsLinearWhiteSpaceChar(quotedText[i]))
+                    continue;
+
+                if (quotedText[i] <= ' ' || quotedText[i] >= 0xFF)
+                    throw new ArgumentException(SR.InvalidParameterValue);
+                else
+                    if (quotedText[i] == '"' &&
+                        (i == 0 || quotedText[i - 1] != '\\'))
+                    throw new ArgumentException(SR.InvalidParameterValue);
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the input character is an allowed character
+        /// Returns false if the input cha----ter is not an allowed character
+        /// </summary>
+        /// <param name="character">input character</param>
+        /// <returns></returns>
+        private static bool IsAllowedCharacter(char character)
+        {
+            //We did not use any of the .Contains methods as
+            //it will result in boxing costs.
+            foreach (char c in s_allowedCharacters)
+            {
+                if (c == character)
+                    return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns true if the input character is an ASCII digit or letter
+        /// Returns false if the input character is not an ASCII digit or letter
+        /// </summary>
+        /// <param name="character">input character</param>
+        /// <returns></returns>
+        private static bool IsAsciiLetterOrDigit(char character)
+        {
+            if (IsAsciiLetter(character))
+            {
+                return true;
+            }
+            if (character >= '0')
+            {
+                return (character <= '9');
+            }
+            return false;
+        }
```

Comment:
Probably just a style thing, but I'd personally have written this whole body as just:
```C#
return IsAsciiLetter(character) || (character >= '0' && character <= '9');
```